### PR TITLE
Challenge 

### PR DESCRIPTION
**Pull request by attania521**

### CODE REVIEW

File: backend/backend/urls.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from django.urls import include, path
+from rest_framework import routers
+from users import views
+
+router = routers.DefaultRouter()
+router.register(r'users', views.UsersViewSet)
+router.register(r'users/?$', views.UsersViewSet)
```
Comment (atomboulian) [inline backend/backend/urls.py:7]: See https://docs.djangoproject.com/en/3.0/ref/settings/#append-slash to cover both cases. Not a big deal though :)

File: backend/users/fixtures/users.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
+[
```
Comment (atomboulian) [inline backend/users/fixtures/users.json:1]: Great that you used fixtures! I love it!

File: backend/users/models.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from django.db import models
+
+class Users(models.Model):
+    id = models.IntegerField(primary_key=True)
```
Comment (atomboulian) [inline backend/users/models.py:4]: Probably want to use an AutoField here to increment appropriately. (Although we used UUIDs and the shortuuid python package to have truly unique IDs from environment to environment.)

File: backend/users/views.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    def get_queryset(self):
+        queryset = self.queryset
+        login_count = self.request.query_params.get('login_count', None)
+        if login_count is not None:
+            queryset = queryset.filter(login_count=login_count)
+
+        login_count_gt = self.request.query_params.get('login_count_gt', None)
+        if login_count_gt is not None:
+            queryset = queryset.filter(login_count__gt=login_count_gt)
```
Comment (atomboulian) [inline backend/users/views.py:17]: These make total sense - nice work here. Another handy library is django-filter to define common filters on the viewset.

File: frontend/src/App.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
```
Comment (atomboulian) [inline frontend/src/App.js:1]: Ah, you created this in React instead of Vue. Very interesting.

File: frontend/src/App.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import ReactDOM from "react-dom";
+import MaterialTable from "material-table";
```
Comment (atomboulian) [inline frontend/src/App.js:3]: Alright, I love that you took a library off the shelf for this.

Comment (atomboulian) [reply frontend/src/App.js:3]: 👍

File: frontend/src/App.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+        }}
+        columns={[
+          { title: "id", field: "id" },
+          { title: "username", field: "username" },
+          { title: "last_login", field: "last_login" },
+          { title: "login_count", field: "login_count" },
+          { title: "project_count", field: "project_count" },
+        ]}
+        data={[
```
Comment (atomboulian) [inline frontend/src/App.js:53]: Oh, so you didn't have your frontend get the data from your backend? Why not?

File: backend/users/serializers.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+from rest_framework_json_api import serializers
+from users.models import Users
+
+class UsersSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = Users
+        fields = ('id', 'username', 'last_login', 'login_count', 'project_count')
```
Comment (atomboulian) [inline backend/users/serializers.py:7]: At least make your `id` a `read_only_field` please.

File: frontend/src/App.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+          {
+            icon: () => <Button variant='contained'>TOGGLE LOGGED IN </Button>,
+            tooltip: 'Filter',
+            onClick: () => setRed(!red),
+            isFreeAction: true
+          },
+          {
+            icon: () => <Button variant='contained' >TOGGLE NEVER LOGGED IN </Button>,
+            tooltip: 'Filter',
+            onClick: () => setGreen(!green),
```
Comment (atomboulian) [inline frontend/src/App.js:25]: Whoops. I think you reversed the colors for the buttons here - should be `setRed(!red)`.

File: frontend/src/App.js (hunk 7 above)
Comment (attania521) [reply frontend/src/App.js:53]: I probably should have wired it up, for some reason didn't do so.  Just thinking now, probably needed to just add a proxy field in my package.json so I don't have to deal with cors and a useCallback hook make an api request.

File: frontend/src/App.js (hunk 5 above)
Comment (attania521) [reply frontend/src/App.js:1]: Yea, kind of just a huge comfort pick. Super comfortable in react, so just went with it for speed/time purposes.

**Pull request closed without merging**